contribs/gnodev/cmd/gnobro/main.go

```diff
@@ -0,0 +1,452 @@
+	)
+}
+
+func execBrowser(cfg *broCfg, args []string, cio commands.IO) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	home := gnoenv.HomeDir()
+
+	var address string
+	var kb keys.Keybase
+	if cfg.defaultAccount != "" {
+		address = cfg.defaultAccount
+
+		var err error
+		kb, err = keys.NewKeyBaseFromDir(home)
+		if err != nil {
+			return fmt.Errorf("unable to load keybase: %w", err)
+		}
+	} else {
+		// create a inmemory keybase
+		kb = keys.NewInMemory()
+		kb.CreateAccount(integration.DefaultAccount_Name, integration.DefaultAccount_Seed, "", "", 0, 0)
+		address = integration.DefaultAccount_Name
+	}
+
+	signer, err := getSignerForAccount(cio, address, kb, cfg)
+	if err != nil {
+		return fmt.Errorf("unable to get signer for account %q: %w", address, err)
+	}
+
+	cl, err := client.NewHTTPClient(cfg.remote)
+	if err != nil {
+		return fmt.Errorf("unable to create http client for %q: %w", cfg.remote, err)
+	}
+
+	gnocl := &gnoclient.Client{
+		RPCClient: cl,
+		Signer:    signer,
+	}
+
+	var path string
+	switch {
+	case len(args) > 0:
+		path = strings.TrimSpace(args[0])
+		path = strings.TrimPrefix(path, gnoPrefix)
+	case cfg.defaultRealm != "":
+		path = strings.TrimLeft(cfg.defaultRealm, gnoPrefix)
+	}
+
+	bcfg := browser.DefaultConfig()
+	bcfg.Readonly = cfg.readonly
+	bcfg.Renderer = lipgloss.DefaultRenderer()
+	bcfg.URLDefaultValue = path
+	bcfg.URLPrefix = gnoPrefix
+	bcfg.URLPrefix = gnoPrefix
+
+	if cfg.sshListener == "" {
+		if cfg.banner {
+			bcfg.Banner = NewBanner_GnoLand()
+		}
+
+		return runLocal(ctx, gnocl, cfg, bcfg, cio)
+	}
+
+	return runServer(ctx, gnocl, cfg, bcfg, cio)
+}
+
+func runLocal(ctx context.Context, gnocl *gnoclient.Client, cfg *broCfg, bcfg browser.Config, io commands.IO) error {
+	var err error
+
+	model := browser.New(bcfg, gnocl)
+	p := tea.NewProgram(model,
+		tea.WithAltScreen(), // use the full size of the terminal in its "alternate screen buffer"
+
+		tea.WithMouseCellMotion(), // turn on mouse support so we can track the mouse wheel
+	)
+
+	devpoint, err := getDevEndpoint(cfg)
+	if err != nil {
+		return fmt.Errorf("unable to parse dev endpoint: %w", err)
+	}
+
+	var wg sync.WaitGroup
+
+	ctx, cancel := context.WithCancelCause(ctx)
+	defer cancel(nil)
+
+	// setup trap signal
+	osm.TrapSignal(func() {
+		cancel(nil)
+		p.Quit()
+	})
+
+	if devpoint != "" {
+		var devcl browser.DevClient
+		devcl.Handler = func(typ events.Type, data any) error {
+			switch typ {
+			case events.EvtReload, events.EvtReset, events.EvtTxResult:
+				p.Send(browser.RefreshRealm())
+			default:
+			}
+
+			return nil
+		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			if err := devcl.Run(ctx, devpoint, nil); err != nil {
+				cancel(fmt.Errorf("dev connection failed: %w", err))
+			}
+		}()
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		_, err := p.Run()
+		cancel(err)
+	}()
+
+	wg.Wait()
+
+	io.Println("Bye!")
+	return context.Cause(ctx)
```

Comment:
I think we should just drop this entirely, it returns `context cancelled` on normal exit, and a status code 1